src/__tests__/waitFor.test.tsx

```diff
@@ -204,6 +204,43 @@ test.each([false, true])(
   }
 );
 
+const blockThread = (timeToBlockThread: number, legacyFakeTimers: boolean) => {
+  jest.useRealTimers();
+  let end = Date.now() + timeToBlockThread;
+
+  while (Date.now() < end) {
+    // do nothing
+  }
+
+  jest.useFakeTimers({ legacyFakeTimers });
+};
+
+test.each([true, false])(
+  'it should not depend on real time when using fake timers (legacyFakeTimers = %s)',
+  async (legacyFakeTimers) => {
+    jest.useFakeTimers({ legacyFakeTimers });
+
+    const mockErrorFn = jest.fn(() => {
+      // Wait 5 seconds so that check time is longer than interval
```

Comment:
The seems to be ms from check the other code.
```suggestion
      // Wait 5 ms so that check time is longer than interval
```